tests/post/features/example_solution.feature

```diff
@@ -0,0 +1,22 @@
+@local @solution
+Feature: Example solution
+    Scenario: Expected Deployment example solution
+        Given the Kubernetes API is available
+        And pods with label 'name=example-solution-operator' are 'Ready'
+        And we have 1 running pod labeled 'name=example-solution-operator' in namespace 'example-environment'
+        When we check the available replicas for 'example-solution-operator' deployment in namespace 'example-environment'
+        Then we have '1' at 'status.available_replicas' for 'example-solution-operator' Deployment in namespace 'example-environment'
+
+    Scenario: Expected Deployments example versionserver
+        Given the Kubernetes API is available
+        And pods with label 'app.kubernetes.io/name=example-versionserver' are 'Ready'
+        And we have 3 running pod labeled 'app.kubernetes.io/name=example-versionserver' in namespace 'example-environment'
+        When we check the available replicas for 'example-versionserver' deployment in namespace 'example-environment'
+        Then we have '3' at 'status.available_replicas' for 'example-versionserver' Deployment in namespace 'example-environment'
+
+    Scenario: Expected HTTP endpoints example versionserver
+        Given the Kubernetes API is available
+        And pods with label 'app.kubernetes.io/name=example-versionserver' are 'Ready'
+        And we have 3 running pod labeled 'app.kubernetes.io/name=example-versionserver' in namespace 'example-environment'
+        When we check the http endpoint '/version' on port '8080' with IP 'spec.cluster_ip' for 'example-versionserver' in namespace 'example-environment'
+        Then service returns '200' status code
```

Comment:
Those are not scenarii, merely static checks that could be in a `Given` or a `Then`.

You want to test the effect of running the CLI `solutions.sh`, build one scenario per command:
- Given no Solution is imported, And the config file is absent, When I run the `solutions.sh import ...` command, Then ...
- Given Solution toto is imported, And it's not marked as active, When I run `solutions.sh activate ...`, Then ...
- etc.

Do the same for creation of environment, adding a solution instance to an environment, and optionally (but really, that's not the point of those tests) check that the Operator works by creating a CR and checking that it deploys a component as expected.

Later iterations will also validate the upgrade of a Solution, but for now, let's focus on those simple actions. Also make sure to clean-up the Solution deployment after the test run (use a `session`-scoped fixture, add the teardown commands to it when you managed to setup something, like importing a solution, and make sure it runs on test completion, be it successful or not).

Reply:
ok